File: N1_4.cpp
#include <iostream>
#include <cmath>
#include <iomanip>
using namespace std;

int main() {
    double x_start, x_end, dx;
    
    cout << "X начальное: ";
    cin >> x_start;
    cout << "X конечное: ";
    cin >> x_end;
    cout << "шаг dx: ";
    cin >> dx;

    cout << "\n" << "    X  |   Y  \n";
    
    // Табулирование
    for (double x = x_start; x <= x_end; x += dx) {
        double y = -sin(M_PI * x / 4.0);
        cout << setw(6) << x << " | " << y << endl;
    }
    
    cout << "\n";
    return 0;
}

File: N1_4.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var xStart, xEnd, dx float64

	fmt.Print("X начальное: ")
	fmt.Scan(&xStart)
	fmt.Print("X конечное: ")
	fmt.Scan(&xEnd)
	fmt.Print("Шаг dx: ")
	fmt.Scan(&dx)

	// Заголовок таблицы
	fmt.Println("   X   |    Y")

	// Табулирование
	for x := xStart; x <= xEnd; x += dx {
		y := -math.Sin(math.Pi * x / 4.0)
		fmt.Printf("%6.1f | %8.4f\n", x, y)
	}

	fmt.Println("-------------------")
}
